intell-media-sdk: build error

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

intel-media-sdk won't build anymore



### Steps to reproduce

build intel-media-sdk

### Expected behaviour

should build

### Screenshots

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/blldmm0jn7c3krgdkilipja7l2d5pbn7-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/zhwg2ap5yyazlab3gganp110139ff9x3-nixos-search-onevplrt-in-run-opengl-driver-lib.patch
patching file api/mfx_dispatch/linux/mfxloader.cpp
patching file tests/unit/suites/mfx_dispatch/linux/mfx_dispatch_test_cases_libs.cpp
applying patch /nix/store/sv8xniq7f86nrcbiv6b1amivhz8wrp58-include-cstdint-explicitly.patch
patching file api/mfx_dispatch/linux/mfxparser.cpp
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LOCALEDIR>
CMake Deprecation Warning at CMakeLists.txt:21 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is GNU 14.3.0
-- The C compiler identification is GNU 14.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /nix/store/67x7pknz0qa2j16x02idf0x98lpcspah-gcc-wrapper-14.3.0/bin/gcc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/67x7pknz0qa2j16x02idf0x98lpcspah-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/67x7pknz0qa2j16x02idf0x98lpcspah-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
...skipping...
In function '_OIter std::transform(_IIter, _IIter, _OIter, _UnaryOperation) [with _IIter = const short unsigned int*; _OIter = unsigned char*; _UnaryOperation = InitPPS(const MfxExtBuffe>
    inlined from 'void InitPPS(const MfxExtBuffer::Param<mfxVideoParam>&, const HEVCEHW::Base::PPS&, VAEncPictureParameterBufferHEVC&)' at /build/source/_studio/mfx_lib/encode_hw/hevc/li>
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_algo.h:4220:12: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
/nix/store/rc5jnpvrc9105bv013bln2q579g7bnfb-libva-2.22.0-dev/include/va/va_enc_hevc.h: In function 'void InitPPS(const MfxExtBuffer::Param<mfxVideoParam>&, const HEVCEHW::Base::PPS&, VAE>
/nix/store/rc5jnpvrc9105bv013bln2q579g7bnfb-libva-2.22.0-dev/include/va/va_enc_hevc.h:623:21: note: at offset [29, 34] into destination object '_VAEncPictureParameterBufferHEVC::row_heig>
  623 |     uint8_t         row_height_minus1[21];
      |                     ^~~~~~~~~~~~~~~~~
In function '_OIter std::transform(_IIter, _IIter, _OIter, _UnaryOperation) [with _IIter = const short unsigned int*; _OIter = unsigned char*; _UnaryOperation = InitPPS(const MfxExtBuffe>
    inlined from 'void InitPPS(const MfxExtBuffer::Param<mfxVideoParam>&, const HEVCEHW::Base::PPS&, VAEncPictureParameterBufferHEVC&)' at /build/source/_studio/mfx_lib/encode_hw/hevc/li>
/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.3.0/include/c++/14.3.0/bits/stl_algo.h:4220:12: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
/nix/store/rc5jnpvrc9105bv013bln2q579g7bnfb-libva-2.22.0-dev/include/va/va_enc_hevc.h: In function 'void InitPPS(const MfxExtBuffer::Param<mfxVideoParam>&, const HEVCEHW::Base::PPS&, VAE>
/nix/store/rc5jnpvrc9105bv013bln2q579g7bnfb-libva-2.22.0-dev/include/va/va_enc_hevc.h:623:21: note: at offset [30, 35] into destination object '_VAEncPictureParameterBufferHEVC::row_heig>
  623 |     uint8_t         row_height_minus1[21];
      |                     ^~~~~~~~~~~~~~~~~
[ 76%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/dumps/dump_mfxla.cpp.o
[ 76%] Building CXX object _studio/mfx_lib/encode_hw/CMakeFiles/encode_hw.dir/hevc/linux/g12/hevcehw_g12_lin.cpp.o
[ 77%] Building CXX object _studio/mfx_lib/encode_hw/CMakeFiles/encode_hw.dir/hevc/linux/g12/hevcehw_g12_rext_lin.cpp.o
[ 77%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/dumps/dump_mfxvp8.cpp.o
[ 77%] Building CXX object _studio/mfx_lib/encode_hw/CMakeFiles/encode_hw.dir/hevc/linux/g12/hevcehw_g12_qp_modulation_lin.cpp.o
[ 78%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/loggers/log.cpp.o
[ 78%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/loggers/log_console.cpp.o
[ 78%] Linking CXX static library ../../../__lib/Release/libencode_hw.a
[ 78%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/loggers/log_etw_events.cpp.o
[ 78%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/loggers/log_file.cpp.o
[ 78%] Built target encode_hw
[ 78%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/loggers/log_syslog.cpp.o
[ 78%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/tracer/tracer.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/tracer/tracer_linux.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/tracer/tracer_windows.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_core.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_core.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_decode.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_enc.cpp.o
[ 79%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_encode.cpp.o
[ 80%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_user.cpp.o
[ 80%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_vpp.cpp.o
[ 80%] Building CXX object tests/unit/suites/tracer/linux/build/CMakeFiles/mfx-tracer.dir/wrappers/mfx_video_fei.cpp.o
[ 80%] Linking CXX shared library ../../../../../../__bin/Release/libmfx-tracer.so
[ 80%] Built target mfx-tracer
make: *** [Makefile:146: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.26, NixOS, 25.05 (Warbler), 25.05.20250508.dda3dcd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/1yvl4w6xbqz33m5xffwy8811lad2gqw4-source`

### Notify maintainers


@midchildan 
@pjungkamp 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/in/intel-media-sdk/package.nix
+++ b/pkgs/by-name/in/intel-media-sdk/package.nix
@@ -42,6 +42,7 @@ stdenv.mkDerivation rec {
     cmake
     pkg-config
   ];
+
   buildInputs = [
     libdrm
     libva
@@ -51,15 +52,14 @@ stdenv.mkDerivation rec {
     libXdmcp
     libpthreadstubs
   ];
-  nativeCheckInputs = [ gtest ];
 
   cmakeFlags = [
     "-DBUILD_SAMPLES=OFF"
-    "-DBUILD_TESTS=${if doCheck then "ON" else "OFF"}"
+    "-DBUILD_TESTS=OFF"
     "-DUSE_SYSTEM_GTEST=ON"
   ];
 
-  doCheck = true;
+  doCheck = false;
 
   meta = {
     description = "Intel Media SDK";

